src/main/java/org/junit/runners/parameterized/BlockJUnit4ClassRunnerWithParameters.java

```diff
@@ -135,7 +138,73 @@ protected void validateFields(List<Throwable> errors) {
 
     @Override
     protected Statement classBlock(RunNotifier notifier) {
-        return childrenInvoker(notifier);
+        Statement statement = childrenInvoker(notifier);
+        statement = withBeforeParams(statement);
+        statement = withAfterParams(statement);
+        return statement;
+    }
+
+    private Statement withBeforeParams(Statement statement) {
+        List<FrameworkMethod> befores = getTestClass()
+                .getAnnotatedMethods(Parameterized.BeforeParam.class);
+        return befores.isEmpty() ? statement : new RunBeforeParams(statement, befores);
+    }
+
+    private class RunBeforeParams extends Statement {
+        private final Statement next;
+        private final List<FrameworkMethod> befores;
+
+        RunBeforeParams(Statement next, List<FrameworkMethod> befores) {
+            this.next = next;
+            this.befores = befores;
+        }
+
+        @Override
+        public void evaluate() throws Throwable {
+            for (FrameworkMethod before : befores) {
+                int paramCount = before.getMethod().getParameterTypes().length;
+                before.invokeExplosively(
+                        null, paramCount == 0 ? (Object[]) null : parameters);
+            }
+            next.evaluate();
+        }
+    }
```

Comment:
Can we reuse `org.junit.internal.runners.statements.RunBefores` and pass `parameters` to it?